app/controllers/documents/TypeController.scala

```diff
@@ -44,26 +47,44 @@ class TypeController @Inject() (
   val controllerComponents: MessagesControllerComponents,
   view: TypeView,
   viewModelProvider: TypeViewModelProvider
-)(implicit ec: ExecutionContext)
+)(implicit ec: ExecutionContext, config: FrontendAppConfig)
     extends FrontendBaseController
     with I18nSupport {
 
   private val prefix: String = "document.type"
 
+  private def findDocTypesNotMaxed(userAnswers: UserAnswers) =
+    userAnswers
+      .get(DocumentsSection)
+      .mapWithIndex {
+        case (_, index) =>
+          userAnswers.get(TypePage(index)).map(_.`type`)
+      }
+      .flatten
+      .filter(allowedDocTypes.contains)
+      .groupBy(identity)
+      .filter(_._2.length < config.maxDocumentsPerType)
+      .keys
+      .toSeq
```

Comment:
Could this logic live in the view model? Pass the document list into the apply method then in the tests you can just mock the response?